### DISCUSSION TITLE
Anti-context enhancement?

### DISCUSSION THREAD

**Discussion by ldave23** [microsoft/presidio#1167] [Q&A]
Hi, I'm aware that Presidio has a context enhancement functionality, which uses context to validate a finding. 

I'm wondering if there is something like an "anti-context", which uses context to invalidate a finding.

For example, consider two examples of sensitive data: "612-345-6789" and "my social security number is 612-345-6789"; while the latter can be validated by context enhancement, the former cannot. In certain situations, it's ideal that both are validated.

The simplest way to recognize both is to simply use regex and avoid context enhancement altogether. However, this introduces actual false positives, for example "my score in the three games were 612-345-6789"; the context here is clearly in a game setting, which is "distant" from the target context of "social security number". Is there some enhancement that only recognizes if a sentence is sufficiently "distant" from the target context? Thanks.

**Comment by omri374** [comment]
Hi, while this isn't supported off-the-shelf, you can achieve it by implementing the `enhance_using_context` of each recognizer. This method would allow you to increase/decrease the score based on any logic, including reducing the score based on words before and after. See more here: https://github.com/microsoft/presidio/blob/93934a93b29f5f3d659725482b22d3bbf4b6e62c/presidio-analyzer/presidio_analyzer/entity_recognizer.py#L91

**Comment by ldave23** [comment]
Got it, I'll need to take a look at how this works.
